streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -81,12 +112,14 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
         "ReceiverTracker", new ReceiverTrackerEndpoint(ssc.env.rpcEnv))
       if (!skipReceiverLaunch) receiverExecutor.start()
       logInfo("ReceiverTracker started")
+      trackerState = Started
     }
   }
 
   /** Stop the receiver execution thread. */
   def stop(graceful: Boolean): Unit = synchronized {
-    if (!receiverInputStreams.isEmpty && endpoint != null) {
+    if (isTrackerStarted) {
+      trackerState = Stopping
```

Comment:
> changing trackerState to Stopping and calling stopReceivers() should be atomic, or it's still possible that StopReceiver won't be sent because the registerReceiver method runs in another thread.

Oh, I see. `ReceiverStarted` will be rejected and force the receiver stop. It's really tricky. Do you think we can solve this issue without  `ReceiverStarted`